src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -136,15 +136,52 @@ public void ProcessStart_TryOpenFolder_UseShellExecuteIsFalse_ThrowsWin32Excepti
             Win32Exception e = Assert.Throws<Win32Exception>(() => Process.Start(new ProcessStartInfo { UseShellExecute = false, FileName = Path.GetTempPath() }));
         }
         
-        [PlatformSpecific(TestPlatforms.Windows)] // Expected behavior varies on Windows and Unix. Refer to #23969
         [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.HasWindowsShell))]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "not supported on UAP")]
         [OuterLoop("Launches File Explorer")]
-        public void ProcessStart_TryOpenFolder_UseShellExecuteIsTrue_DoesNotThrow()
+        public void ProcessStart_UseShellExecuteTrue_OpenMissingFile_Throws()
         {
-            using (var px = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = Path.GetTempPath() }))
+            string fileToOpen = Path.Combine(Environment.CurrentDirectory, "_no_such_file.TXT");
+            Win32Exception e = Assert.Throws<Win32Exception>(() => Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen }));
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.HasWindowsShell)), InlineData(true), InlineData(false)]
```

Comment:
nit, doesn't matter this time, we don't combine attributes, so this would be
```c#
[ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.HasWindowsShell))]
[InlineData(true)]
[InlineData(false)]
```